// @quadrata/kyb-react/lib/esm/store/features/form/form.slice.js
var _a;
import { createSlice } from '@reduxjs/toolkit';
export var FormField;
(function (FormField) {
    // Intro
    FormField["BUSINESS_NAME"] = "BUSINESS_NAME";
    // Section 1
    FormField["EMAIL"] = "EMAIL";
    FormField["BUSINESS_PHONE"] = "BUSINESS_PHONE";
    FormField["BUSINESS_WEBSITE_URLS"] = "BUSINESS_WEBSITE_URLS";
    FormField["COMPANY_LEGAL_NAME"] = "COMPANY_LEGAL_NAME";
    FormField["COMPANY_DBA"] = "COMPANY_DBA";
    FormField["COMPANY_FORMATION_DATE"] = "COMPANY_FORMATION_DATE";
    FormField["BUSINESS_DESCRIPTION"] = "BUSINESS_DESCRIPTION";
    // Section 2
    FormField["JURISDICTION_OF_FORMATION"] = "JURISDICTION_OF_FORMATION";
    FormField["PRINCIPAL_PLACE_OF_BUSINESS"] = "PRINCIPAL_PLACE_OF_BUSINESS";
    FormField["COUNTRIES_OF_BUSINESS_ACTIVITY"] = "COUNTRIES_OF_BUSINESS_ACTIVITY";
    FormField["BUSINESS_TAX_ID"] = "BUSINESS_TAX_ID";
    FormField["BUSINESS_REGISTRATION_NUMBER"] = "BUSINESS_REGISTRATION_NUMBER";
    FormField["BUSINESS_NAICS_CODE"] = "BUSINESS_NAICS_CODE";
    // Section 3
    FormField["BENEFICIAL_OWNERS"] = "BENEFICIAL_OWNERS";
    FormField["CONTROLLING_PERSON_LEGAL_NAME"] = "CONTROLLING_PERSON_LEGAL_NAME";
    FormField["CONTROLLING_PERSON_TITLE"] = "CONTROLLING_PERSON_TITLE";
    FormField["CONTROLLING_PERSON_EMAIL"] = "CONTROLLING_PERSON_EMAIL";
    // Section 4
    FormField["WALLET_ADDRESSES"] = "WALLET_ADDRESSES";
    FormField["TEST_WALLET_ADDRESSES"] = "TEST_WALLET_ADDRESSES";
    // Tos
    FormField["SIGNATURE"] = "SIGNATURE";
})(FormField || (FormField = {}));
export var FormFileType;
(function (FormFileType) {
    // User upload files
    FormFileType["CERTIFICATE_OF_INCORPORATION"] = "CERTIFICATE_OF_INCORPORATION";
    FormFileType["BYLAWS"] = "BYLAWS";
    FormFileType["KYC_AML_PROGRAM"] = "KYC_AML_PROGRAM";
    FormFileType["AUTHORIZED_SIGNATORY_LIST"] = "AUTHORIZED_SIGNATORY_LIST";
    FormFileType["BANK_VERIFICATION"] = "BANK_VERIFICATION";
    // Auto generated file containing all form fields
    FormFileType["FORM_FIELDS"] = "FORM_FIELDS";
})(FormFileType || (FormFileType = {}));
var generateInitialState = function () {
    var _a, _b;
    return ({
        fields: (_a = {},
            // Intro
            _a[FormField.BUSINESS_NAME] = '',
            // Section 1
            _a[FormField.EMAIL] = '',
            _a[FormField.BUSINESS_PHONE] = '',
            _a[FormField.BUSINESS_WEBSITE_URLS] = '',
            _a[FormField.COMPANY_LEGAL_NAME] = '',
            _a[FormField.COMPANY_DBA] = '',
            _a[FormField.COMPANY_FORMATION_DATE] = '',
            _a[FormField.BUSINESS_DESCRIPTION] = '',
            // Section 2
            _a[FormField.JURISDICTION_OF_FORMATION] = '',
            _a[FormField.PRINCIPAL_PLACE_OF_BUSINESS] = '',
            _a[FormField.COUNTRIES_OF_BUSINESS_ACTIVITY] = '',
            _a[FormField.BUSINESS_TAX_ID] = '',
            _a[FormField.BUSINESS_REGISTRATION_NUMBER] = '',
            _a[FormField.BUSINESS_NAICS_CODE] = '',
            // Section 3
            _a[FormField.BENEFICIAL_OWNERS] = '',
            _a[FormField.CONTROLLING_PERSON_LEGAL_NAME] = '',
            _a[FormField.CONTROLLING_PERSON_TITLE] = '',
            _a[FormField.CONTROLLING_PERSON_EMAIL] = '',
            // Section 4
            _a[FormField.WALLET_ADDRESSES] = '',
            _a[FormField.TEST_WALLET_ADDRESSES] = '',
            // ToS
            _a[FormField.SIGNATURE] = '',
            _a),
        files: (_b = {},
            // User upload files
            _b[FormFileType.CERTIFICATE_OF_INCORPORATION] = undefined,
            _b[FormFileType.BYLAWS] = undefined,
            _b[FormFileType.KYC_AML_PROGRAM] = undefined,
            _b[FormFileType.AUTHORIZED_SIGNATORY_LIST] = undefined,
            _b[FormFileType.BANK_VERIFICATION] = undefined,
            // Auto generated files
            _b[FormFileType.FORM_FIELDS] = undefined,
            _b),
    });
};
var formSlice = createSlice({
    name: 'form-slice',
    initialState: generateInitialState(),
    reducers: {
        setField: function (state, action) {
            var _a = action.payload, key = _a.key, value = _a.value;
            state.fields[key] = value;
        },
        setFile: function (state, action) {
            var _a = action.payload, key = _a.key, value = _a.value;
            if (value) {
                state.files[key] = value;
            }
            else if (!value && state.files[key] && state.files[key].objectURL) {
                // Revoking object URL to prevent memory leaks
                URL.revokeObjectURL(state.files[key].objectURL);
                // Removing from store
                state.files[key] = undefined;
            }
            else {
                // Falling through, Removing from store
                state.files[key] = undefined;
            }
        },
        reset: function (state) {
            state = generateInitialState();
            return state;
        },
    },
});
export var setField = (_a = formSlice.actions, _a.setField), setFile = _a.setFile, reset = _a.reset;
export default formSlice.reducer;
