// @quadrata/kyb-react/lib/esm/src/store/features/navigation/navigation.slice.js
var _a;
import { createSlice } from '@reduxjs/toolkit';
export var Page;
(function (Page) {
    Page["INTRO"] = "INTRO";
    Page["TOS"] = "TOS";
    Page["SIGN_WALLET"] = "SIGN_WALLET";
    Page["VERIFF"] = "VERIFF";
    Page["VERIFF_STATUS"] = "VERIFF_STATUS";
    Page["KYC_FORM"] = "KYC_FORM";
    Page["MINT_NFT"] = "MINT_NFT";
    Page["MINT_NFT_STATUS"] = "MINT_NFT_STATUS";
    Page["MINT_NFT_COMPLETE"] = "MINT_NFT_COMPLETE";
})(Page || (Page = {}));
export var PageError;
(function (PageError) {
    PageError["AUTH_ERROR"] = "AUTH_ERROR";
    PageError["VERIFF_ERROR"] = "VERIFF_ERROR";
    PageError["KYC_ERROR"] = "KYC_ERROR";
})(PageError || (PageError = {}));
var generateInitialState = function () { return ({
    page: Page.INTRO,
    error: false,
}); };
var navigationSlice = createSlice({
    name: 'navigation-slice',
    initialState: generateInitialState(),
    reducers: {
        setPage: function (state, action) {
            if (state.page !== action.payload) {
                state.page = action.payload;
                state.error = action.payload in PageError;
            }
        },
        reset: function (state) {
            state = generateInitialState();
            return state;
        },
    },
});
export var setPage = (_a = navigationSlice.actions, _a.setPage), reset = _a.reset;
export default navigationSlice.reducer;
