// @quadrata/kyb-react/lib/esm/store/features/navigation/navigation.slice.js
var _a;
import { createSlice } from '@reduxjs/toolkit';
export var PageKyb;
(function (PageKyb) {
    // Intro page
    PageKyb["INTRO"] = "INTRO";
    // Business Application Section Pages
    PageKyb["BASIC_INFORMATION"] = "BUSINESS_INFORMATION";
    PageKyb["TAXES"] = "TAXES";
    PageKyb["OWNERSHIP"] = "OWNERSHIP";
    PageKyb["WALLETS"] = "WALLETS";
    PageKyb["DOCUMENTS"] = "DOCUMENTS";
    PageKyb["TOS"] = "TOS";
    // State Pages
    PageKyb["UPLOAD"] = "UPLOAD";
    PageKyb["COMPLETE"] = "COMPLETE";
    // Error Pages
    PageKyb["ERROR"] = "ERROR";
    PageKyb["AUTH_ERROR"] = "AUTH_ERROR";
})(PageKyb || (PageKyb = {}));
var generateInitialState = function () { return ({
    page: PageKyb.INTRO,
    error: false,
}); };
var navigationSlice = createSlice({
    name: 'navigation-slice',
    initialState: generateInitialState(),
    reducers: {
        setPage: function (state, action) {
            if (state.page !== action.payload) {
                state.page = action.payload;
            }
        },
        reset: function (state) {
            state = generateInitialState();
            return state;
        },
    },
});
export var setPage = (_a = navigationSlice.actions, _a.setPage), reset = _a.reset;
export default navigationSlice.reducer;
