// @quadrata/kyb-react/lib/esm/src/store/store.js
import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector } from 'react-redux';
import authReducer from './features/auth/auth.slice';
import kycFormReducer from './features/kyc/kyc-form.slice';
import settingsReducer from './features/settings/settings.slice';
import navigationReducer from './features/navigation/navigation.slice';
import localizationReducer from './features/localization/localization.slice';
import veriffStatusReducer from './features/veriff/veriff-status.slice';
import veriffSessionReducer from './features/veriff/veriff-session.slice';
import veriffResponseReducer from './features/veriff/veriff-response.slice';
import ethereumContractReducer from './features/ethereum/ethereum-contract.slice';
export var store = configureStore({
    reducer: {
        auth: authReducer,
        kycForm: kycFormReducer,
        settings: settingsReducer,
        navigation: navigationReducer,
        localization: localizationReducer,
        pollingStatus: veriffStatusReducer,
        veriffSession: veriffSessionReducer,
        veriffResponse: veriffResponseReducer,
        ethereumContract: ethereumContractReducer,
    },
});
// Use throughout your app instead of plain `useDispatch` and `useSelector`
export var useAppDispatch = function () { return useDispatch(); };
export var useAppSelector = useSelector;
