// @quadrata/kyb-react/lib/esm/src/store/features/veriff/veriff-session.slice.js
var _a;
import { createSlice } from '@reduxjs/toolkit';
var generateInitialState = function () { return ({
    id: '',
    url: '',
    vendorData: '',
    host: '',
    status: '',
    sessionToken: '',
}); };
export var veriffSessionSlice = createSlice({
    name: 'veriff-session',
    initialState: generateInitialState(),
    reducers: {
        updateVeriffSession: function (state, action) {
            state.id = action.payload.id;
            state.url = action.payload.url;
            state.vendorData = action.payload.vendorData;
            state.host = action.payload.host;
            state.status = action.payload.status;
            state.sessionToken = action.payload.sessionToken;
        },
        clearVeriffSessionSlice: function (state) {
            state.id = '';
            state.url = '';
            state.vendorData = '';
            state.host = '';
            state.status = '';
            state.sessionToken = '';
        },
        reset: function (state) {
            state = generateInitialState();
            return state;
        },
    },
});
export var updateVeriffSession = (_a = veriffSessionSlice.actions, _a.updateVeriffSession), clearVeriffSessionSlice = _a.clearVeriffSessionSlice, reset = _a.reset;
export default veriffSessionSlice.reducer;
