// @quadrata/kyb-react/lib/esm/src/store/features/veriff/veriff-status.slice.js
var _a;
import { createSlice } from '@reduxjs/toolkit';
import { IDVerificationStatus } from '../../../controllers/api/getKycStatus.api';
export var AMLVerificationStatus;
(function (AMLVerificationStatus) {
    AMLVerificationStatus["PENDING"] = "PENDING";
    AMLVerificationStatus["LOW_RISK"] = "LOW_RISK";
    AMLVerificationStatus["MEDIUM_RISK"] = "MEDIUM_RISK";
    AMLVerificationStatus["HIGH_RISK"] = "HIGH_RISK";
    AMLVerificationStatus["REVIEW_REQUIRED"] = "REVIEW_REQUIRED";
})(AMLVerificationStatus || (AMLVerificationStatus = {}));
var generateInitialState = function () { return ({
    amlVerificationStatus: AMLVerificationStatus.PENDING,
    idVerificationStatus: IDVerificationStatus.PENDING,
    blockChainMineComplete: false,
}); };
export var slice = createSlice({
    name: 'PollingStatusSlice',
    initialState: generateInitialState(),
    reducers: {
        updateIdVerificationStatus: function (state, action) {
            state.idVerificationStatus = action.payload;
        },
        updateAmlVerificationStatus: function (state, action) {
            state.amlVerificationStatus = action.payload;
        },
        blockChainMineSetComplete: function (state) {
            state.blockChainMineComplete = true;
        },
        clearKYCPollingStatus: function (state) {
            state.amlVerificationStatus = AMLVerificationStatus.PENDING;
            state.blockChainMineComplete = false;
        },
        clearIdPollingStatus: function (state) {
            state.idVerificationStatus = IDVerificationStatus.PENDING;
        },
        clearPollingStatusSlice: function (state) {
            state.amlVerificationStatus = AMLVerificationStatus.PENDING;
            state.blockChainMineComplete = false;
            state.idVerificationStatus = IDVerificationStatus.PENDING;
        },
        reset: function (state) {
            state = generateInitialState();
            return state;
        },
    },
});
export var updateIdVerificationStatus = (_a = slice.actions, _a.updateIdVerificationStatus), updateAmlVerificationStatus = _a.updateAmlVerificationStatus, blockChainMineSetComplete = _a.blockChainMineSetComplete, clearKYCPollingStatus = _a.clearKYCPollingStatus, clearIdPollingStatus = _a.clearIdPollingStatus, clearPollingStatusSlice = _a.clearPollingStatusSlice, reset = _a.reset;
export default slice.reducer;
